From 4186c94104ee528abd6860611b49515f3e6ec644 Mon Sep 17 00:00:00 2001
From: Thomas Guillem <thomas@gllm.fr>
Date: Fri, 18 Oct 2019 10:34:07 +0200
Subject: [PATCH] input/item: sort before attaching slaves

The order of readdir() is completely filesystem dependent.

Slaves should be attached using the same order across multiple OSses/access
modules.

This fixes the test_libvlc_slaves on some system when the entry order from
readdir() is different.

This patch need to be backported to VLC 3.0.
---
 src/input/item.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/input/item.c b/src/input/item.c
index e8db39ca28..fbd74cc95f 100644
--- a/src/input/item.c
+++ b/src/input/item.c
@@ -1809,8 +1809,8 @@ void vlc_readdir_helper_finish(struct vlc_readdir_helper *p_rdh, bool b_success)
 {
     if (b_success)
     {
-        rdh_attach_slaves(p_rdh, p_rdh->p_node);
         rdh_sort(p_rdh->p_node);
+        rdh_attach_slaves(p_rdh, p_rdh->p_node);
     }
     free(p_rdh->psz_ignored_exts);
 
-- 
2.20.1

